lib/spack/spack/build_environment.py

```diff
@@ -125,12 +127,22 @@ def __init__(self, name, jobs):
         self.jobs = jobs
 
     def __call__(self, *args, **kwargs):
+        """parallel, and jobs_env from kwargs are swallowed and used here;
+        remaining arguments are passed through to the superclass.
+        """
+
         disable = env_flag(SPACK_NO_PARALLEL_MAKE)
-        parallel = not disable and kwargs.get('parallel', self.jobs > 1)
+        parallel = (not disable) and kwargs.pop('parallel', self.jobs > 1)
 
         if parallel:
-            jobs = "-j%d" % self.jobs
-            args = (jobs,) + args
+            args = ('-j{0}'.format(self.jobs),) + args
+            jobs_env = kwargs.pop('jobs_env', None)
+            if jobs_env:
+                # Caller wants us to set an environment variable to
+                # control the parallelism.
+                make_env = kwargs.pop('env', []).copy()
```

Comment:
This looks strange to me comparing to the next line: it looks like `make_env` can be a list. If `jobs_env` is a string, it looks like the next line will fail (attempt to index a list with a string). Should the default be `{}`?

Reply:
The default should be `{}`, correct. Apparently the existing tests don't attempt to run the tests for any CMake packages with parallelism enabled.

Reply:
For this sort of error (and to just cover the lines of code) you could add a simple test to the `make_executable` python test module (where `jobs_env` is set).